cabby/geo/walk.py

```diff
@@ -627,11 +625,17 @@ def generate_and_save_rvs_routes(self,
       columns=['osmid', 'geometry', 'main_tag'])
 
     counter = 0
+    attempt = 0
     while counter < n_samples:
       entity = self.get_single_sample(map)
       if entity is None:
+        attempt += 1
+        if attempt >= MAX_NUM_FAILED:
+          sys.exit("Reached max number of failed attempts.")
         continue
+      attempt = 0 
       counter += 1
+      logging.info("Created sample {}/{}".format(counter,n_samples))
```

Comment:
It's generally shorter and easier to read when using f'' instead, e.g.:

```
logging.info(f"Created sample {counter}/{n_samples}".)
```

Can you switch to this style here and in future?